src/charm.py

```diff
@@ -1,88 +1,133 @@
 #
 # Learn more at: https://juju.is/docs/sdk
 
+"""A Juju charm for Identity Platform Login UI."""
 import logging
 
 from charms.traefik_k8s.v1.ingress import (
+    IngressPerAppReadyEvent,
     IngressPerAppRequirer,
+    IngressPerAppRevokedEvent,
 )
-from charms.observability_libs.v0.kubernetes_service_patch import KubernetesServicePatch
-from ops.charm import CharmBase
+from ops.charm import CharmBase, ConfigChangedEvent, HookEvent, WorkloadEvent
 from ops.main import main
-from ops.model import ActiveStatus, MaintenanceStatus
+from ops.model import (ActiveStatus, BlockedStatus, MaintenanceStatus,
+                       ModelError, WaitingStatus)
+from ops.pebble import ChangeError, Layer
 
-# Log messages can be retrieved using juju debug-log
 logger = logging.getLogger(__name__)
 
-VALID_LOG_LEVELS = ["info", "debug", "warning", "error", "critical"]
 
-
-class KratosUiOperatorCharm(CharmBase):
-    """Charm the service."""
+class IdentityPlatformLoginUiOperatorCharm(CharmBase):
+    """Charmed Identity Platform Login UI."""
 
     def __init__(self, *args):
+        """Initialize Charm."""
         super().__init__(*args)
-        self._container_name = "kratos-ui"
-        self.service_patcher = KubernetesServicePatch(
-            self, [("api", 3000)]
-        )
+        self._container_name = "login_ui"
+        self._container = self.unit.get_container(self._container_name)
+        self._application_port = "8080"
+
+        self.unit.open_port("tcp", int(self._application_port))
         self.ingress = IngressPerAppRequirer(
             self,
             relation_name="ingress",
-            port=3000,
+            port=self._application_port,
             strip_prefix=True,
         )
 
-        self.framework.observe(self.on.kratos_ui_pebble_ready, self._on_kratos_ui_pebble_ready)
+        self.framework.observe(self.on.login_ui_pebble_ready,
+                               self._on_login_ui_pebble_ready)
         self.framework.observe(self.on.config_changed, self._on_config_changed)
         self.framework.observe(self.ingress.on.ready, self._on_ingress_ready)
+        self.framework.observe(self.ingress.on.revoked,
+                               self._on_ingress_revoked)
+
+    def _on_login_ui_pebble_ready(self, event: WorkloadEvent) -> None:
+        """Define and start a workload using the Pebble API."""
+        self._handle_status_update_config(event)
+
+    def _on_config_changed(self, event: ConfigChangedEvent) -> None:
+        """Handle changed configuration."""
+        self._handle_status_update_config(event)
+
+    def _handle_status_update_config(self, event: HookEvent) -> None:
+        if not self._container.can_connect():
+            event.defer()
+            logger.info("""Cannot connect to Login_UI container.
+ Deferring the event.""")
+            self.unit.status = WaitingStatus(
+                "Waiting to connect to Login_UI container"
+            )  # noqa:E501
```

Comment:
is this `noqa` really needed? It seems the newline makes the line short enough

Reply:
this was a result of formatting with black. All #noqa directives has been deleted.